src/components/Popover/PopoverContent.tsx

```diff
@@ -0,0 +1,157 @@
+}
+
+interface StyledPopoverProps extends React.CSSProperties {
+    isShown: boolean;
+    transform?: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    translate?: any;
+    top?: string | number | undefined;
+    left?: string | number | undefined;
+    bottom?: string | number | undefined;
+    right?: string | number | undefined;
+    onKeyDown: (event: React.KeyboardEvent<HTMLDivElement>) => void;
+    ref: MutableRefObject<HTMLDivElement>;
+    tabIndex: number;
+    role: string;
+}
+
+const StyledPopoverParent = styled.div.attrs({ theme })<StyledPopoverProps>`
+    ${({ isShown, transform, top, left, bottom, right, position }) => css`
+        box-sizing: border-box;
+        position: ${position};
+        top: ${top};
+        left: ${left};
+        bottom: ${bottom};
+        right: ${right};
+        transform: ${transform};
+        transition: ${mapTransitions(['opacity'], '0.15s', 'ease-in-out')};
+        width: auto;
+        height: auto;
+        z-index: 1000;
+        box-shadow: ${get('shadows.small')};
+        max-height: none;
+        opacity: ${isShown ? '1' : '0'};
+
+        &:focus {
+            outline: 0;
+        }
+    `}
+`;
+
+const PopoverContentContainer = styled(Card)`
+    display: block;
+    padding: ${Spaces[2]};
+`;
+
+export const PopoverContent = ({
+    children,
+    isShown,
+    onClose,
+    offset,
+    placement,
+    triggerRef
+}: PopoverContentProps): JSX.Element => {
+    const popoverRef = React.useRef<HTMLDivElement | null>(null);
+
+    const { styles, update } = usePopper(triggerRef, popoverRef.current, {
+        placement,
+        strategy: 'fixed',
+        modifiers: [
+            {
+                name: 'offset',
+                enabled: Boolean(offset),
+                options: {
+                    offset: [0, offset]
+                }
+            },
+            {
+                name: 'flip',
+                enabled: true
+            }
+        ]
+    });
+
+    const { popper } = styles;
+
+    React.useEffect(() => {
+        // On each update we need to set focus on the popover content, so that we can close the Popover content with ESC key
+        const timer = setTimeout(() => {
+            popoverRef?.current?.focus();
+        }, 100);
+
+        // eslint-disable-next-line no-console
+        if (update)
+            update()
+                .then((): void => undefined)
+                // eslint-disable-next-line no-console
+                .catch(error => console.log('Error updating Popover position -', error));
+
+        return () => {
+            clearTimeout(timer);
+        };
+    }, [update]);
+
+    // Close the Popover content with click on ESC key
+    const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+        if (event.keyCode === 27 && onClose) onClose(event);
+    };
+
+    // Hide the Popover content when user clicks outside
+    useClickOutside(popoverRef, event => {
```

Comment:
have you checked if it works correctly when I click not on the popover ref, but on its children?

For example `ref` is attached to a `div` container with a markup. If I hit the markup (e.g. `<p>` content) the target of the event will be the paragraph, not ref. Hence, you can close the hint when it is not supposed to close.

Reply:
yes I tested it worked correctly, also this is moved in next commits :)